packages/engine/Source/Widget/CesiumWidget.js

```diff
@@ -387,6 +442,94 @@ function CesiumWidget(container, options) {
+
+    let dataSourceCollection = options.dataSources;
+    let destroyDataSourceCollection = false;
+    if (!defined(dataSourceCollection)) {
+      dataSourceCollection = new DataSourceCollection();
+      destroyDataSourceCollection = true;
+    }
+
+    const dataSourceDisplay = new DataSourceDisplay({
+      scene: scene,
+      dataSourceCollection: dataSourceCollection,
+    });
+
+    const eventHelper = new EventHelper();
+    this._dataSourceChangedListeners = {};
+    this._automaticallyTrackDataSourceClocks = defaultValue(
+      options.automaticallyTrackDataSourceClocks,
+      true
+    );
+
+    this._dataSourceCollection = dataSourceCollection;
+    this._destroyDataSourceCollection = destroyDataSourceCollection;
+    this._dataSourceDisplay = dataSourceDisplay;
+    this._eventHelper = eventHelper;
+
+    eventHelper.add(this._clock.onTick, CesiumWidget.prototype._onTick, this);
+    eventHelper.add(
+      scene.morphStart,
+      CesiumWidget.prototype._clearTrackedObject,
+      this
+    );
+
+    //Listen to data source events in order to track clock changes.
+    eventHelper.add(
+      dataSourceCollection.dataSourceAdded,
+      CesiumWidget.prototype._onDataSourceAdded,
+      this
+    );
+    eventHelper.add(
+      dataSourceCollection.dataSourceRemoved,
+      CesiumWidget.prototype._onDataSourceRemoved,
+      this
+    );
+
+    eventHelper.add(scene.postRender, CesiumWidget.prototype._postRender, this);
+
+    // We need to subscribe to the data sources and collections so that we can clear the
+    // tracked object when it is removed from the scene.
+    // Subscribe to current data sources
+    const dataSourceLength = dataSourceCollection.length;
+    for (let i = 0; i < dataSourceLength; i++) {
+      this._dataSourceAdded(dataSourceCollection, dataSourceCollection.get(i));
+    }
+    this._dataSourceAdded(undefined, dataSourceDisplay.defaultDataSource);
+
+    // Hook up events so that we can subscribe to future sources.
+    eventHelper.add(
+      dataSourceCollection.dataSourceAdded,
+      CesiumWidget.prototype._dataSourceAdded,
+      this
+    );
+    eventHelper.add(
+      dataSourceCollection.dataSourceRemoved,
+      CesiumWidget.prototype._dataSourceRemoved,
+      this
+    );
+
+    // Subscribe to left clicks and zoom to the picked object.
+    function pickAndTrackObject(e) {
+      const entity = pickEntity(that, e);
+      if (defined(entity)) {
+        //Only track the entity if it has a valid position at the current time.
+        if (
+          Property.getValueOrUndefined(entity.position, that.clock.currentTime)
+        ) {
+          that.trackedEntity = entity;
+        } else {
+          that.zoomTo(entity);
+        }
+      } else if (defined(that.trackedEntity)) {
+        that.trackedEntity = undefined;
+      }
+    }
+
+    this._screenSpaceEventHandler.setInputAction(
+      pickAndTrackObject,
+      ScreenSpaceEventType.LEFT_DOUBLE_CLICK
+    );
```

Comment:
I think this click handler should be moved back to the `Viewer`. Moving the _functionality_ to track an entity into the `CesiumWidget` is good so you can do `CesiumWidget.trackedEntity = ...` but setting up the click handler interaction is a `Viewer` oriented feature. Basically I would expect the handler in `Viewer` to be able to do `this._cesiumWidget.trackedEntity = pickEntity()` (with the extra checks it previously had).

Reply:
I agree that it was quite strange to have the click handled in the viewer and the double click handled in the widget, but the intent was to have the tracking feature activated with double click available within the widget without further user coding.
However this is still possible through the API without too much effort, so yes, let's bring it back!

Reply:
> However this is still possible through the API without too much effort, so yes, let's bring it back!

Exactly, I think it's good to expand the widget API with this functionality, making it easier for users to add _if they want_ but still only do it for them in the viewer